tools/lambda-promtail/lambda-promtail/s3.go

```diff
@@ -27,6 +27,28 @@ var (
 
 	// regex that extracts the timestamp (RFC3339) from message log
 	timestampRegex = regexp.MustCompile(`\w+ (?P<timestamp>\d+-\d+-\d+T\d+:\d+:\d+\.\d+Z)`)
+
+	// regex that matches the format of a single line of an AWS Load Balancer log
+	// https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-access-logs.html
+	// https://docs.aws.amazon.com/athena/latest/ug/application-load-balancer-logs.html
+	loadBalancerLogLineRegex = regexp.MustCompile(`(?P<type>[^ ]*) (?P<time>[^ ]*) (?P<elb>[^ ]*) (?P<client_ip>[^ ]*):(?P<client_port>[0-9]*) (?P<target_ip>[^ ]*)[:-](?P<target_port>[0-9]*) (?P<request_processing_time>[-.0-9]*) (?P<target_processing_time>[-.0-9]*) (?P<response_processing_time>[-.0-9]*) (?P<elb_status_code>|[-0-9]*) (?P<target_status_code>-|[-0-9]*) (?P<received_bytes>[-0-9]*) (?P<sent_bytes>[-0-9]*) \"(?P<request_verb>[^ ]*) (?P<request_url>.*) (?P<request_proto>- |[^ ]*)\" \"(?P<user_agent>[^\"]*)\" (?P<ssl_cipher>[A-Z0-9-_]+) (?P<ssl_protocol>[A-Za-z0-9.-]*) (?P<target_group_arn>[^ ]*) \"(?P<trace_id>[^\"]*)\" \"(?P<domain_name>[^\"]*)\" \"(?P<chosen_cert_arn>[^\"]*)\" (?P<matched_rule_priority>[-.0-9]*) (?P<request_creation_time>[^ ]*) \"(?P<actions_executed>[^\"]*)\" \"(?P<redirect_url>[^\"]*)\" \"(?P<lambda_error_reason>[^ ]*)\" \"(?P<target_port_list>[^\s]+?)\" \"(?P<target_status_code_list>[^\s]+)\" \"(?P<classification>[^ ]*)\" \"(?P<classification_reason>[^ ]*)\"`)
+
+	// the indexes of the fields in the log that we want to create labels for
+	typeIndex              = loadBalancerLogLineRegex.SubexpIndex("type")
+	targetPortIndex        = loadBalancerLogLineRegex.SubexpIndex("target_port")
```

Comment:
Not sure port is useful here - any particular use-case we need it for?

Reply:
My original thought was that we could sort traffic by http, etc. but it looks like the "type" label will give us that same functionality. Removed the target port label in the new revision.